app/components/TaskList/index.tsx

```diff
@@ -121,215 +121,269 @@ const TaskList: React.FC<Props> = ({ projectRevision, mode }) => {
     summary: useShowGrowthbookFeature("teimp") ? 8 : 7,
   };
 
-  return (
-    <div className="container">
-      <h2>
-        {mode === "view"
-          ? projectByProjectId.proposalReference
-          : mode === "update"
-          ? "Editing: " + projectByProjectId.proposalReference
-          : "Add a Project"}
-      </h2>
-      <ol>
-        {/* Project Overview Section */}
-        <TaskListSection
-          defaultExpandedState={
-            currentStep === "0" ||
-            projectOverviewStatus === ATTENTION_REQUIRED_STATUS
-          }
-          listItemNumber="1"
-          listItemName="Project Overview"
-        >
-          <TaskListItem
-            stepName="0"
-            linkUrl={getProjectRevisionFormPageRoute(id, 0)}
-            formTitle="Project overview"
-            formStatus={projectOverviewStatus}
-            currentStep={currentStep}
-            mode={mode}
-          />
-        </TaskListSection>
-
-        {/* Project Details Section */}
-        <TaskListSection
-          defaultExpandedState={
-            currentStep === "1" ||
-            currentStep === "2" ||
-            projectManagersStatus === ATTENTION_REQUIRED_STATUS ||
-            projectContactsStatus === ATTENTION_REQUIRED_STATUS
-          }
-          listItemNumber="2"
-          listItemName="Project Details"
-          listItemMode={mode === "update" ? "" : "(optional)"}
-        >
-          <TaskListItem
-            stepName="1"
-            linkUrl={getProjectRevisionFormPageRoute(id, 1)}
-            formTitle="Project managers"
-            formStatus={projectManagersStatus}
-            currentStep={currentStep}
-            mode={mode}
-          />
-          <TaskListItem
-            stepName="2"
-            linkUrl={getProjectRevisionFormPageRoute(id, 2)}
-            formTitle="Project contacts"
-            formStatus={projectContactsStatus}
-            currentStep={currentStep}
-            mode={mode}
-          />
-        </TaskListSection>
+  const projectOverviewSection = (step: string) => {
+    return (
+      <TaskListSection
+        defaultExpandedState={
+          currentStep === "0" ||
```

Comment:
These steps are still hardcoded in the different components. What we want to achieve with this refactor is to make everything generated, including links and navigation, so we can easily reorder, add, remove or hide items without breaking magic numbers in different places.
There is definitely a bit of research to be done on how to do that, I'm happy to talk about it one of these days to find out what the best path would be for the implementation

Reply:
Agreed. Tried a few different approaches and didn't get a good outcome (passing step, and currentStep into the config).  Discussing this would be good.